component/grant-type/src/main/java/org/wso2/carbon/identity/oauth2/grant/jwt/JWTBearerGrantHandler.java

```diff
@@ -104,6 +107,38 @@ public void init() throws IdentityOAuth2Exception {
         }
     }
 
+    /**
+     * get resident Identity Provider
+     *
+     * @param tenantDomain tenant Domain
+     * @param jwtIssuer    issuer extracted from assertion
+     * @return resident Identity Provider
+     * @throws IdentityOAuth2Exception
+     */
+    private IdentityProvider getIdentityProvider(String tenantDomain, String jwtIssuer) throws IdentityOAuth2Exception {
+        String issuer = StringUtils.EMPTY;
+        IdentityProvider residentIdentityProvider = null;
+        try {
+            residentIdentityProvider = IdentityProviderManager.getInstance().getResidentIdP(tenantDomain);
+        } catch (IdentityProviderManagementException e) {
+            String errorMsg = String.format(ERROR_GET_RESIDENT_IDP, tenantDomain);
+            throw new IdentityOAuth2Exception(errorMsg, e);
+        }
+        FederatedAuthenticatorConfig[] fedAuthnConfigs = residentIdentityProvider.getFederatedAuthenticatorConfigs();
+        FederatedAuthenticatorConfig oauthAuthenticatorConfig =
+                IdentityApplicationManagementUtil.getFederatedAuthenticator(fedAuthnConfigs,
+                        IdentityApplicationConstants.Authenticator.OIDC.NAME);
+        if (oauthAuthenticatorConfig != null) {
+            issuer = IdentityApplicationManagementUtil.getProperty(oauthAuthenticatorConfig.getProperties(),
+                    OPENID_IDP_ENTITY_ID).getValue();
+        }
+        if (jwtIssuer.equals(issuer)) {
+            return residentIdentityProvider;
+        } else {
+            return null;
+        }
```

Comment:
Returning `null` here means we have to `null` check in another place. IMO we should take advantage of `Optional` in java 8. Furthermore, this code can be changed to a single line as shown below:
```java
return jwtIssuer.equals(issuer) ? Optional.ofNullable(residentIdentityProvider) : Optional.empty();
```
WDYT?